packages/edit-site/src/components/add-new-template/new-template.js

```diff
@@ -128,12 +128,9 @@ export default function NewTemplate( {
 			setCanvasMode( 'edit' );
 
 			// Navigate to the created template editor.
-			window.queueMicrotask( () => {
```

Comment:
These were needed previously because we had two synchronous changes causing effects in hooks that update the URL (canvas change and template navigation) and these two changes to the "history" path were overriding each other, the micro task ensured that they were done one after the other. 

Now that the url is the source of truth of everything, it's way simpler.